Common/Data/Consolidators/PeriodCountConsolidatorBase.cs

```diff
@@ -89,6 +89,17 @@ protected PeriodCountConsolidatorBase(Func<DateTime, CalendarInfo> func)
             _period = Time.OneSecond;
         }
 
+        /// <summary>
+        /// Creates a consolidator to produce a new <typeparamref name="TConsolidated"/> instance representing the last count pieces of data or the period, whichever comes first
+        /// </summary>
+        /// <param name="funcpyobj">Python function object that defines the start time of a consolidated data</param>
+        protected PeriodCountConsolidatorBase(PyObject funcpyobj)
+        {
```

Comment:
Why not just:
```csharp
        protected PeriodCountConsolidatorBase(PyObject funcpyobj)
            : this(funcpyobj.ConvertToDelegate<Func<DateTime, CalendarInfo>>())
        {
        }
```

Reply:
Modified it now . Added overloaded constructors to derived classes from `PeriodCountConsolidatorBase`.
Still not added unit test. Was trying to find correct place to do that.

Reply:
Added unit tests.